route/build/src/process_ntopo.f90

```diff
@@ -221,13 +229,83 @@ subroutine ntopo(&
 
  endif  ! if writing the data
 
+ ! ---------- temporary code: populate old data structures --------------------------------------------------
+
+ ! allocate space
+ allocate(RPARAM(nSeg), NETOPO(nSeg), stat=ierr)
+ if(ierr/=0)then; message=trim(message)//'unable to allocate space for old data structures'; return; endif
+
+ ! loop through stream segments
+ do iSeg=1,nSeg
+
+  ! print progress
+  if(mod(iSeg,100000)==0) print*, 'Copying to the old data structures: iSeg, nSeg = ', iSeg, nSeg
+
+  ! ----- reach parameters -----
+
+  ! copy data into the reach parameter structure
+  RPARAM(iSeg)%RLENGTH = structSEG(iSeg)%var(ixSEG%length)%dat(1)
+  RPARAM(iSeg)%R_SLOPE = structSEG(iSeg)%var(ixSEG%slope)%dat(1)
+  RPARAM(iSeg)%R_MAN_N = structSEG(iSeg)%var(ixSEG%width)%dat(1)
+  RPARAM(iSeg)%R_WIDTH = structSEG(iSeg)%var(ixSEG%man_n)%dat(1)
+
+  ! compute variables
+  RPARAM(iSeg)%BASAREA = sum(structSEG(iSeg)%var(ixSEG%hruArea)%dat)
+
+  ! NOT USED: UPSAREA is not currently used, but could be useful...
+  !        --> can loop through all upstream basins (in reach_list?)
+  RPARAM(iSeg)%UPSAREA = realMissing  ! upstream area (zero if headwater basin)
+  RPARAM(iSeg)%TOTAREA = realMissing  ! UPSAREA + BASAREA
```

Comment:
a few problems if they are missing are 1) discharge conversion in (5g) printing progress in route_runoff.f90, and 2) computing goodBas (which is gone now) in kwt_route.f90.